### ISSUE TITLE
Has CoreOS stable dropped kernel support for SMB 3.0?

### ISSUE THREAD

**Issue by drbolsen** [coreos/bugs#1387]
Hey there,

We are currently running CoreOS 899.17.0 (stable) on Azure cloud with a drop-in mount.cifs that allows us to mount Azure File shares. In the current state we can successfully mount Azure File shares with smb 3.0 settings straight away, e.g.

`sudo mount.cifs //myaccountname.file.core.windows.net/mysharename ./mymountpoint -o vers=3.0,username=myaccountname,password=StorageAccountKeyEndingIn==,dir_mode=0777,file_mode=0777`

A couple of days ago we upgraded one of our boxes to a new stable version 1010.5.0 and immediately noticed that Azure File share was not working with the following error trail in the logs:

```
Jun 02 21:54:08 core-03 kernel: Key type cifs.idmap registered
Jun 02 21:54:08 core-03 kernel: CIFS VFS: SMB3 encryption not supported yet
Jun 02 21:54:08 core-03 kernel: CIFS VFS: SMB3 encryption not supported yet
Jun 02 21:54:08 core-03 kernel: CIFS VFS: encrypted frame parsing not supported yet
Jun 02 21:54:08 core-03 kernel: CIFS VFS: Bad protocol string signature header 424d53fd
Jun 02 21:54:08 core-03 kernel: CIFS VFS: Bad SMB detected. The Mid=96665694354
Jun 02 21:54:08 core-03 kernel: 00000000: 7d000000 424d53fd 39805a79 5434b049  ...}.SMByZ.9I.4T
Jun 02 21:54:08 core-03 kernel: 00000010: 54c9bc5b 339b4e1a 975a5d0b 81b97092  [..T.N.3.]Z..p..
Jun 02 21:54:08 core-03 kernel: 00000020: 00000016 00000000 00000049 00010000  ........I.......
Jun 02 21:54:08 core-03 kernel: CIFS VFS: No task to wake, unknown frame received! NumMids 1
Jun 02 21:54:08 core-03 kernel: 00000000: 7d000000 424d53fd 39805a79 5434b049  ...}.SMByZ.9I.4T
Jun 02 21:54:08 core-03 kernel: 00000010: 54c9bc5b 339b4e1a 975a5d0b 81b97092  [..T.N.3.]Z..p..
Jun 02 21:54:08 core-03 kernel: 00000020: 00000016 00000000 00000049 00010000  ........I.......
Jun 02 21:54:08 core-03 kernel: 00000030: 7c000051 8200d800 7ff4f67b c122c5a4  Q..|....{.....".
Jun 02 21:54:08 core-03 kernel: 00000040: 610301a7                             ...a
Jun 02 21:56:14 core-03 kernel: CIFS VFS: Server myaccountname.file.core.windows.net has not responded in 120 seconds. Reconnecting...
Jun 02 21:56:14 core-03 kernel: CIFS VFS: cifs_mount failed w/return code = -11
```

Rolling back to the previous stable version resolved the issue immediately.

Wonder if somehow SMB 3.0 support in the kernel was dropped in the latest builds of CoreOS?

Thanks!

**Comment by aemr3** [comment]
I have the same problem. CoreOS version is 1010.5.0.

> Jun 08 01:44:21 core3 kernel: CIFS VFS: SMB3 encryption not supported yet
Jun 08 01:44:21 core3 kernel: CIFS VFS: SMB3 encryption not supported yet
Jun 08 01:44:21 core3 kernel: CIFS VFS: encrypted frame parsing not supported yet
Jun 08 01:44:21 core3 kernel: CIFS VFS: Bad protocol string signature header 424d53fd
Jun 08 01:44:21 core3 kernel: 00000000: 7d000000 424d53fd 617453f9 61e2e12d  ...}.SMB.Sta-..a
Jun 08 01:44:21 core3 kernel: 00000010: 92008c7c 3807b63f a57241de 19fd5392  |...?..8.Ar..S..
Jun 08 01:44:21 core3 kernel: 00000020: 0000001a 00000000 00000049 00010000  ........I.......
Jun 08 01:44:21 core3 kernel: 00000000: 7d000000 424d53fd 617453f9 61e2e12d  ...}.SMB.Sta-..a
Jun 08 01:44:21 core3 kernel: 00000010: 92008c7c 3807b63f a57241de 19fd5392  |...?..8.Ar..S..
Jun 08 01:44:21 core3 kernel: 00000020: 0000001a 00000000 00000049 00010000  ........I.......
Jun 08 01:44:21 core3 kernel: 00000030: 74000d81 8fe04000 4bd9e8eb 53756d27  ...t.@.....K'muS
Jun 08 01:44:21 core3 kernel: 00000040: d2e6f067                             g...

**Comment by aemr3** [comment]
@drbolsen You can use vers=2.1 until this is fixed.

**Comment by drbolsen** [comment]
Thanks mate, but we have a requirement for traffic encryption so it is easier for us to stay with previous stable version of CoreOS. 

BTW I went on checking kernel flags in relation to SMB for 1010.5 it appears that there is no difference from  899.17. I am currently puzzled why is this error.

**Comment by sxmichael** [comment]
Having same issue here, vers=2.1 really works, but need 3.0 (due to encryption and cross-region)

**Comment by mjg59** [comment]
@drbolsen As far as I can tell, the kernel has never had support for encrypted SMB3.0 shares. Are you certain that encryption was ever working previously?

**Comment by aemr3** [comment]
@mjg59 yes it was working

**Comment by mjg59** [comment]
@emre tcpdump showed the packets were encrypted?

**Comment by aemr3** [comment]
I am not so sure about it but I am sure that I was using version 3.0 before upgrading.

**Comment by mjg59** [comment]
The kernel has always claimed that SMB3 encryption is unsupported, so my suspicion is that it was failing silently and falling back to unencrypted content previously. I'll set up a test environment here.

**Comment by drbolsen** [comment]
Hmmm, not good. I checked the traffic - apparently it is not encrypted.
 
`17:10:17.471759 IP (tos 0x0, ttl 123, id 14920, offset 0, flags [DF], proto TCP (6), length 149)
    file.ml1prdstr01a.store.core.windows.net.microsoft-ds > csh-core-02.client.cloud.46064: Flags [P.], cksum 0x8b36 (correct), seq 17453:17550, ack 32125, win 16381, options [nop,nop,TS val 3041154 ecr 945795479], length 97 SMB-over-TCP packet:(raw data or continuation?)

	0x0000:  4500 0095 3a48 4000 7b06 3b14 bfef c012  E...:H@.{.;.....
	0x0010:  0a00 0005 01bd b3f0 a716 fa3f 8cee 4ba9  ...........?..K.
	0x0020:  8018 3ffd 8b36 0000 0101 080a 002e 6782  ..?..6........g.
	0x0030:  385f b197 0000 005d fe53 4d42 4000 0100  8_.....].SMB@...
	0x0040:  0000 0000 0800 0100 0900 0000 0000 0000  ................
	0x0050:  8f03 0000 0000 0000 f670 0000 0100 0000  .........p......
	0x0060:  7500 0000 00d4 8080 556d 5b9b d59c 7ba3  u.......Um[...{.
	0x0070:  8fa8 b7b9 1baf 2436 1100 5000 0d00 0000  ......$6..P.....
	0x0080:  0000 0000 0000 0000 6d79 7365 6372 6574  ........mysecret
	0x0090:  6461 7461 0a                             data.`

**Comment by mjg59** [comment]
@drbolsen https://github.com/Azure/azure-content/blob/master/articles/storage/storage-how-to-use-files-linux.md indicates that this is correct, unfortunately.

**Comment by drbolsen** [comment]
Thanks mate. Yeah, we spent sometime yesterday digging the issue. It appears that CIFS tools (we were using the latest cifs-utils 6.5) are lagging support for encryption option, however from kernel perspective it should be there.

I haven't tried it on CoreOS yet but I have found quite few comments about people using smbclient with -e (--encrypt) option successfully. smbclient is useless for our use case still it means that it should be possible to encrypt SMB traffic with a modern kernel. 

I guess it is still a mystery why the behaviour of kernel has changed and why the previous version was falling back to 2.1 silently.
